Client/Frontend/Browser/TabManager.swift

```diff
@@ -54,6 +54,8 @@ class TabManager : NSObject {
     private let defaultNewTabRequest: NSURLRequest
     private let navDelegate: TabManagerNavDelegate
     private var configuration: WKWebViewConfiguration
+    private var networkActivityCount = 0 // Counts the number of tabs with network activity
+    private var tabsNetworkActivity = [WKWebView: Bool]()
```

Comment:
Please get rid of `tabsNetworkActivity`. It introduces a huge memory leak since you are holding on to `WKWebView` instances forever. And it is not actually used in any way to start/stop the network activity spinner. I think having the counter is enough.